src/requirementsFileExercise/requirements_file_exercise.py

```diff
@@ -1,32 +1,42 @@
 from typing import List
 
-from src.requirementsFileExercise.data_extractor import get_file_name, get_package, has_requirement_file_reference
+from src.requirementsFileExercise.package import Package
 from src.requirementsFileExercise.file_reader import read_file_lines
 
 
 def create_libraries_string(file_name: str) -> str:
     libraries: List[str] = []
-    files: List[str] = [file_name]
+    requirements_files: List[str] = [file_name]
     scanned_files: List[str] = []
 
-    for requirement_file in files:
-        if requirement_file not in scanned_files:
-            text: List[str] = read_file_lines(requirement_file)
+    for requirements_file in requirements_files:
+        if requirements_file not in scanned_files:
+            text: List[str] = read_file_lines(requirements_file)
             file_libraries, files_references = _create_libraries_list(text)
             libraries += file_libraries
-            files += files_references
-            scanned_files.append(requirement_file)
+            requirements_files += files_references
+            scanned_files.append(requirements_file)
 
-    return ' '.join(libraries)
+    return ' '.join([str(library) for library in libraries])
+
+
+def _add_package(libraries: List[Package], package: Package):
+    for library in libraries:
+        if library.name == package.name:
+            if library.version != package.version:
+                library.version = max(library.version, package.version)
+                return
+    libraries.append(package)
 
 
 def _create_libraries_list(lines: List[str]) -> List[str]:
     files: List[str] = []
     libraries: List[str] = []
     for line in lines:
-        line: str = line.strip()
-        if has_requirement_file_reference(line):
-            files.append(get_file_name(line))
-        elif get_package(line) not in libraries:
-            libraries.append(get_package(line))
+        file_name: str = Package.get_file_name(line)
```

Comment:
just for you to understand what is meant to be consistent for me:

In the other defs you called to the 'file' -->requirment_file and now is just a file.. better you name it 'requirment_file_name'